credoai/lens/lens_validation.py

```diff
@@ -34,52 +36,68 @@ def check_model_data_consistency(model, data):
     # check predict
     # Keras always outputs numpy types (not Tensor or something else)
     if "predict" in model.__dict__.keys() and data.y is not None and data.X is not None:
-        try:
-            mini_pred, batch_size = check_prediction_model_output(model.predict, data)
-            if not mini_pred.size:
-                # results for all presently supported models are ndarray results
-                raise Exception("Empty return results from predict function.")
-            if isinstance(data.y, np.ndarray) and (
-                mini_pred.shape != data.y[:batch_size].shape
-            ):
-                raise Exception("Predictions have mismatched shape from provided y")
-            if isinstance(data.y, pd.Series) and (
-                mini_pred.shape != data.y.head(batch_size).shape
-            ):
-                raise Exception("Predictions have mismatched shape from provided y")
-        except Exception as e:
-            raise ValidationError(
-                "Lens.model predictions do not match expected form implied by provided labels y.",
-                e,
-            )
+        if isinstance(model.model_like, (DummyClassifier, DummyRegression)):
```

Comment:
Doesn't seem great to have validation dependent on Dummy class. Why can't we use the same functionality as the rest of the validation? `predict` and `predict_proba` should work identically to other models.

Reply:
Right now for normal models, I'm doing `predict` on some small chunk of `data.X` to avoid too much heavy computation. I then compare shapes/sizes of predictions and `data.y`

For `DummyClassifier.predict(X)`, you just get the whole prediction array, whereas `data.y` is still subsetted.

One solution would be to do predictions for all data (no batching). The problem in that case is that image data would take forever.
Another solution is to set `batch_size = len(data.y)` for Dummy models.

Reply:
Alright I reworked this a little bit. 

The only place batch was actually used (once set) was to compare overall shapes (n_samples, output1, output2, ...).

In the reworking, I check that the per-sample shape is the same (so check that the (output1, output2, ...) part is the same shape). We'll implicitly assume n_samples in predictions vs. `data.y` are the same here and rely on downstream library functions (e.g. `sklearn.accuracy_score`) to throw an error if the user provided mismatched sample sizes for `data.y` and DummyClassifier (note that it's not possible for this to be an issue with vanilla ClassificationModel because that uses `data.X` and `data.y` which have to be the same).

Maybe there's an even better solution that pushes this into the DummyClassifier??